### ISSUE TITLE
Default ISO Zoned DateTime format only works for multiples of 100 Milliseconds

### ISSUE THREAD

**Issue by joshbode** [JuliaTime/TimeZones.jl#41]
I had expected that `string` and the `AbstractString` constructor for `ZonedDateTime` would be inverses of one another, e.g. `x == ZonedDateTime(string(x))` as is the case for `Date` and `DateTime`.

However, it only appears to work for `ZonedDateTime`s where the milliseconds are a multiple of 100.

```julia
julia> using TimeZones

julia> tz = TimeZone("Australia/Melbourne")
Australia/Melbourne (UTC+10/UTC+11)

julia> x = ZonedDateTime(now(), tz)
2016-12-28T10:00:50.316+11:00

julia> string(x)
"2016-12-28T10:00:50.316+11:00"

julia> ZonedDateTime(string(x))
ERROR: ArgumentError: Unrecognized time zone: 16+11:00
 in TimeZones.FixedTimeZone(::String) at /Users/josh/.julia/v0.5/TimeZones/src/types.jl:86
 in slotparse(::Base.Dates.DelimitedSlot{Base.Dates.TimeZone}, ::String, ::String) at /Users/josh/.julia/v0.5/TimeZones/src/io.jl:64
 in getslot(::String, ::Base.Dates.DelimitedSlot{Base.Dates.TimeZone}, ::String, ::Int64) at ./dates/io.jl:160
 in parse(::String, ::Base.Dates.DateFormat) at ./dates/io.jl:175
 in TimeZones.ZonedDateTime(::String) at /Users/josh/.julia/v0.5/TimeZones/src/io.jl:94
```

However, if I truncate to multiples of 100 milliseconds, it works:
```julia
julia> y = trunc(x, Second) + Millisecond(trunc(Int64(Millisecond(x)) / 100) * 100)
2016-12-28T10:00:50.3+11:00

julia> ZonedDateTime(string(y))
2016-12-28T10:00:50.3+11:00
```

## Version Info
```julia
julia> versioninfo()
Julia Version 0.5.0
Commit 3c9d753* (2016-09-19 18:14 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin15.6.0)
  CPU: Intel(R) Core(TM)2 Duo CPU     P7550  @ 2.26GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Penryn)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, penryn)

julia> Pkg.status("TimeZones")
 - TimeZones                     0.2.5
```

**Comment by joshbode** [comment]
Looks like `string` uses the following format:
```julia
DateFormat("yyyy-mm-ddTHH:MM:SS.ssszzz")
```
while `ZonedDateTime` uses `TimeZones.ISOZonedDateTimeFormat` with only one `s`, i.e.
```julia
DateFormat("yyyy-mm-ddTHH:MM:SS.szzz")
```

**Comment by omus** [comment]
Parsing needs some additional work when it comes to dealing with timezones. For example right now the default `DateFormat` requires that milliseconds are provided when parsing. For `DateTime` parsing these can be excluded.

**Comment by omus** [comment]
Addressed by #42

**Comment by joshbode** [comment]
True - in the meantime, I'm doing this:

```julia
function Base.convert(::Type{TimeZones.ZonedDateTime}, x::AbstractString)
    tz = TimeZones.FixedTimeZone(x[end-5:end])
    x = x[1:end-6]
    if contains(x, ".")
        x, ms = rsplit(x, '.'; limit=2)
        ms = parse(Int64, ms)
    else
        ms = 0
    end
    t = Libc.strptime("%Y-%m-%dT%H:%M:%S", x)
    ZonedDateTime(DateTime(t.year + 1900, t.month + 1, t.mday, t.hour - 1, t.min, t.sec, ms), tz)
end
```

Note: I'm using `strptime` because the `DateTime` parser is slow at the moment.
